### PR TITLE
Implement `$limit` aggregation pipeline stage

### LINKED ISSUE #1425: Implement `$limit` aggregation pipeline stage
https://www.mongodb.com/docs/manual/reference/operator/aggregation/limit/

DOD:
- [x] Test `{$limit: x}` on collection that contains more than x document. (May need to use `$sort` stage before `$limit` to ensure consistent order)
- [x] Test with `$match` stage (again may need `$sort`)
- [x] Test with `$group` stage (again may need `$sort`)
- [x] any other case

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #1425.

## Readiness checklist

* [ ] I added/updated unit tests.
* [x] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [x] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by noisersup** [comment]
> @noisersup this pull request has merge conflicts.

Didn't know that mergify does that!

File: integration/aggregate_compat_test.go [hunk 1]
```diff
@@ -712,6 +736,114 @@ func TestAggregateCompatGroupCount(t *testing.T) {
...
+			},
+			resultType: emptyResult,
+		},
+		"Int64Overflow": {
+			pipeline: bson.A{
+				bson.D{{"$sort", bson.D{{"_id", -1}}}},
+				bson.D{{"$limit", float64(1 << 86)}},
+			},
+			resultType: emptyResult,
+		},
```
Comment (noisersup) [inline integration/aggregate_compat_test.go:813]: We pass a float here, as it's the only possible way (apart from decimal128) to pass the number higher than max long

File: internal/handlers/common/params.go [hunk 2]
```diff
@@ -133,6 +135,11 @@ func GetWholeNumberParam(value any) (int64, error) {
 			return 0, errInfinity
 		}
 
+		if value > float64(math.MaxInt64) ||
+			value < float64(math.MinInt64) {
+			return 0, errLongExceeded
+		}
+
```
Comment (noisersup) [inline internal/handlers/common/params.go:142]: There was a bug in the `GetWholeNumberParam` function, that returned negative `int64` values for positive `float64` exceeding `int64` range.

File: internal/handlers/common/params.go [hunk 3]
```diff
@@ -147,6 +154,48 @@ func GetWholeNumberParam(value any) (int64, error) {
...
+		return 0, commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrStageLimitInvalidArg,
+			fmt.Sprintf("invalid argument to $limit stage: Cannot represent as a 64-bit integer: $limit: %#v", value),
+		)
+	default:
+		panic(fmt.Sprint("Unexpected error: ", err))
+	}
+
+	switch {
+	case max < 0:
```
Comment (chilagrow) [inline internal/handlers/common/params.go:184]:
```suggestion
	case limit < 0:
```
Just so it's easier to read :hugs:

File: internal/handlers/common/params.go (hunk 2 above)
Comment (chilagrow) [reply internal/handlers/common/params.go:142]: wow nice catch!

**Review by chilagrow — COMMENTED** [review]
Looks great! Just a few minor comments :)

**Comment by chilagrow** [comment]
Please also checkout this PR https://github.com/FerretDB/FerretDB/pull/2254 it has changes on limit

File: internal/handlers/common/params.go [hunk 4]
```diff
@@ -147,6 +154,48 @@ func GetWholeNumberParam(value any) (int64, error) {
 	}
 }
 
+// GetLimitStageParam returns $limit stage argument from the provided value.
+// It returns the proper error if value doesn't meet requirements.
+func GetLimitStageParam(value any) (int64, error) {
```
Comment (AlekSi) [inline internal/handlers/common/params.go:159]: I added `GetLimitParam` in #2254; let's merge that other PR first, then see if it is possible to re-use it

Comment (noisersup) [reply internal/handlers/common/params.go:159]: `GetLimitStageParam` already handles errors to match message. Also from my understanding, `$limit` stage could have different requirements, for example `find`'s limit can be 0, but `$limit` cannot.

**Review by w84thesun — APPROVED** [review]
Nice job! Added a few totally optional naming suggestions.

File: internal/handlers/common/params.go [hunk 5]
```diff
@@ -147,6 +154,48 @@ func GetWholeNumberParam(value any) (int64, error) {
...
+			commonerrors.ErrStageLimitInvalidArg,
+			fmt.Sprintf("invalid argument to $limit stage: Expected an integer: $limit: %#v", value),
+		)
+	case errors.Is(err, errLongExceeded):
+		return 0, commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrStageLimitInvalidArg,
+			fmt.Sprintf("invalid argument to $limit stage: Cannot represent as a 64-bit integer: $limit: %#v", value),
+		)
+	default:
+		panic(fmt.Sprint("Unexpected error: ", err))
```
Comment (rumyantseva) [inline internal/handlers/common/params.go:180]: I think it's a little bit too heavy to panic here, shouldn't we return a `lazyerror`? This way, we can trace where the error came from if there is one.

**Review by rumyantseva — COMMENTED** [review]
Great job overall! 

I agree with Dmitry's and Chi's comments and left one more comment.

File: internal/handlers/common/params.go (hunk 4 above)
Comment (AlekSi) [reply internal/handlers/common/params.go:159]: I don't think we should maintain two separate functions that do almost the same job. From your description, it sounds like we could move some checks to `GetLimitParam`, call it from `GetLimitStageParam` and add a check for 0.

File: internal/handlers/common/params.go [hunk 6]
```diff
@@ -147,6 +154,48 @@ func GetWholeNumberParam(value any) (int64, error) {
 	}
 }
 
+// GetLimitStageParam returns $limit stage argument from the provided value.
+// It returns the proper error if value doesn't meet requirements.
+func GetLimitStageParam(value any) (int64, error) {
+	max, err := GetWholeNumberParam(value)
```
[the remark this answers is not in the document]
Comment (noisersup) [reply internal/handlers/common/params.go:160]: nice catch

File: Taskfile.yml [hunk 7]
```diff
@@ -5,7 +5,7 @@ env:
   GORACE: halt_on_error=1,history_size=2
 
 vars:
-  INTEGRATIONTIME: 20m
+  INTEGRATIONTIME: 22m
```
Comment (AlekSi) [inline Taskfile.yml:8]: Please update it to 25m to match #2246

**Review by chilagrow — APPROVED** [review]

**Review by rumyantseva — DISMISSED** [review]
LGTM

**Review by Fashander — APPROVED** [review]

File: internal/handlers/common/params.go (hunk 5 above)
Comment (noisersup) [reply internal/handlers/common/params.go]: fixed

File: internal/handlers/common/params.go (hunk 4 above)
Comment (noisersup) [reply internal/handlers/common/params.go:160]: If I understand our final agreement, we stay with separate function for a while and find a better way later, right?

File: internal/handlers/common/params.go [hunk 8]
```diff
@@ -147,6 +154,48 @@ func GetWholeNumberParam(value any) (int64, error) {
...
+
+	switch {
+	case err == nil:
+	case errors.Is(err, errUnexpectedType):
+		return 0, commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrStageLimitInvalidArg,
+			fmt.Sprintf("invalid argument to $limit stage: Expected a number in: $limit: %#v", value),
+		)
```
[the remark this answers is not in the document]
Comment (noisersup) [reply internal/handlers/common/params.go:170]: Done

**Review by chilagrow — COMMENTED** [review]
There is conflict

**Review by rumyantseva — DISMISSED** [review]

**Review by w84thesun — DISMISSED** [review]

**Review by chilagrow — APPROVED** [review]

**Review by Fashander — APPROVED** [review]

File: internal/handlers/common/params.go (hunk 4 above)
Comment (AlekSi) [reply internal/handlers/common/params.go:160]: That's fine for now

**Pull request merged**